### PR TITLE
Akimbo delay applies to autofire

### PR DESCRIPTION
**Pull request by Lumipharon**
## About The Pull Request
When firing guns akimbo, there is an extra fire delay associated with it that means you shoot a bit slower. Currently this only applies to semi auto firing, not to full auto.
Makes autofire actually respect the akimbo_additional_delay var.

Also reduced the extra delay value for all SMGs to 0.2 from 0.5. This means guns that fire at 0.15 will fire at 0.2 akimbo.
Rifles reduced to 0 as they're terrible at akimbo, and don't need to be even worse.

## Why It's Good For The Game
Akimbo additional delay now actually works with autofiring. This has balance implications for guns that are used akimbo with autofire :^)

## Changelog
:cl:
fix: Autofiring while akimbo now actually shoots slower like semi auto
balance: Reduced akimbo additional delay for SMG's and rifles to compensate
/:cl:

### CODE REVIEW

**Review by BraveMole — CHANGES_REQUESTED** [review]
akimbo_additional_delay is a multiplier

**Comment by Lumipharon** [comment]
> akimbo_additional_delay is a multiplier

I realised this yesterday, but testing is weird.
If it was adding 0.5, delay it should be 0.65 (like a dmr) but it definitely isn't that slow. It does however seem to shoot slower than one handed... code, how do you work.

File: code/modules/projectiles/gun_system.dm [hunk 1]
```diff
@@ -652,7 +652,7 @@
 	set_target(null)
 	windup_checked = WEAPON_WINDUP_NOT_CHECKED
 	if (dual_wield == TRUE)
-		modify_fire_delay(-akimbo_additional_delay) // Removes the additional delay from auto_fire
+		modify_fire_delay(-(fire_delay - fire_delay/(1 + akimbo_additional_delay))) // Removes the additional delay from auto_fire
```
Comment (BraveMole) [inline code/modules/projectiles/gun_system.dm:655]:
```suggestion
		modify_fire_delay(-fire_delay + fire_delay/(1 + akimbo_additional_delay)) // Removes the additional delay from auto_fire
```

**Review by BraveMole — CHANGES_REQUESTED** [review]

**Review by BraveMole — CHANGES_REQUESTED** [review]
ok for 0,2 for t90, but what about the other rifles/mg? they should stay at zero, it's not like they are any good

**Comment by Lumipharon** [comment]
> ok for 0,2 for t90, but what about the other rifles/mg? they should stay at zero, it's not like they are any good

Default is still 0.5, but does anyone actually use any rifles akimbo? You need RC + Gyro to fire straight, and you still have terrible accuracy.
I'm happy to add lower values in case people do though.

File: code/modules/projectiles/gun_system.dm [hunk 2]
```diff
@@ -650,7 +651,9 @@
 	shots_fired = 0//Let's clean everything
 	set_target(null)
 	windup_checked = WEAPON_WINDUP_NOT_CHECKED
-	dual_wield = FALSE
+	if (dual_wield == TRUE)
+		modify_fire_delay(-fire_delay + fire_delay/(1 + akimbo_additional_delay)) // Removes the additional delay from auto_fire
```
Comment (TiviPlus) [inline code/modules/projectiles/gun_system.dm:655]:
```suggestion
	if(dual_wield)
		modify_fire_delay(-fire_delay + fire_delay/(1 + akimbo_additional_delay)) // Removes the additional delay from auto_fire
```

Comment (Lumipharon) [reply code/modules/projectiles/gun_system.dm:655]: Neat

**Review by BraveMole — APPROVED** [review]

**Pull request merged**